ccmlib/scylla_repository.py

```diff
@@ -69,22 +69,23 @@ def release_packages(s3_url, version, arch='x86_64', scylla_product='scylla'):
     :param version:
     :return:
     """
-    major_version = version if version.count('.') == 1 else '.'.join(n for n in version.split('.')[:2])
-    s3_url = s3_url.format(major_version)
 
+    major_version = extract_major_version(version)
+    s3_url = s3_url.format(major_version)
+    
     files_in_bucket = aws_bucket_ls(s3_url)
 
     if not files_in_bucket:
         raise RuntimeError(
             f"Failed to get release packages list for version {major_version}. URL: {s3_url}.")
 
     all_unified_packages = [name for name in files_in_bucket if f'{scylla_product}-{arch}-unified-package' in name]
-    candidates = [package for package in all_unified_packages if ".rc" not in package and '.rc' not in version]
+    candidates = [package for package in all_unified_packages if any([f'{c}rc' not in package and f'{c}rc' not in version for c in ['.', '-', '~']])]
```

Comment:
you want all to mismatch, no?  not any...
anyhow, why not use a regular expression to match `[.~-]rc`?

Reply:
actually found out, the whole logic isn't needed, since we later scan this list for the matching version.
I think it's intend not to fetch release candidate when asking for something like `release:5.1` which has only RCs.

I think it's point less, if someone asks for a release that in process it's o.k. to receive the release candidate